### PR TITLE
Initial sketch of code to compute and apply fixes

### PR DESCRIPTION
**Pull request by bwilkerson**
Untested and incomplete. Just to make sure we're on the same page for the direction we're heading.

### CODE REVIEW

**Comment by devoncarew** [comment]
In your PR, can you remove the `stable` line here: https://github.com/dart-lang/dart2_fix/blob/master/.travis.yml#L10

File: lib/src/deprecation_fix.dart [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+   * The resource provider used to access the file system.
+   */
+  final ResourceProvider resourceProvider;
+
+  DeprecationFixer(this.resourceProvider);
+
+  Future<void> fixFiles(List<String> paths) async {
+    ContextLocator locator =
+        new ContextLocator(resourceProvider: resourceProvider);
+    List<AnalysisContext> contexts =
```
Comment (devoncarew) [inline lib/src/deprecation_fix.dart:26]: wow, this is ridiculously terse

File: lib/src/deprecation_fix.dart [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+class DeprecationFixer {
+  /**
+   * The resource provider used to access the file system.
+   */
+  final ResourceProvider resourceProvider;
+
+  DeprecationFixer(this.resourceProvider);
+
+  Future<void> fixFiles(List<String> paths) async {
```
Comment (devoncarew) [inline lib/src/deprecation_fix.dart:23]: We could also limit to one path here, but either way is fine with me.

**Comment by bwilkerson** [comment]
Done

**Comment by devoncarew** [comment]
If it helps, what I'm thinking for the UI:

```
> dart2_fix

Checking project foo_bar...

lib/main.dart
  line 23, deprecated Duration.ZERO
  line 37, deprecated Duration.ZERO
lib/src/foo.dart.dart
  line 23, deprecated Duration.ZERO

37 changes found.
```

and then:

```
dart2fix --apply

lib/main.dart
   7 changes Duration.ZERO ==> Duration.zero
lib/src/foo.dart
   1 change Duration.BAZ ==> Duration.baz
   3 changes Duration.ZERO ==> Duration.zero

37 changes total.
```

In other words, a run of the tool w/o args will just indicate what changes it's found; a run of the tool with something like `--apply` would actually perform the changes on disk (and indicate approx. what changed).

**Comment by devoncarew** [comment]
I would remove lines https://github.com/dart-lang/dart2_fix/blob/master/.travis.yml#L22 and 23 as well - the stable option, and the the dartfmt task.

**Review by devoncarew — APPROVED** [review]
Thanks! lgtm once green

**Comment by bwilkerson** [comment]
Also done

**Comment by bwilkerson** [comment]
That sounds like a reasonable UI. I'll refactor to support it.

**Pull request merged**